### ISSUE TITLE
Xcode 15 and iOS 17 crashes on Ably projects

### ISSUE THREAD

**Issue by flaviodsilverio** [ably/ably-cocoa#1804]
### Which version of the Ably SDK are you using? 
1.2.20

### On which platform does the issue happen? 
iOS 17

### Are you using Carthage? 
No

### Are you using Cocoapods? 
Yes, 1.12.1.

### Which version of Xcode are you using? 
(`xcodebuild -version`)

### What did you do?
After installing Xcode 15, just trying to run any app (we have 2 using ably) will make it crash as soon as it starts. I figured, and fixed, the issue, locally, however it would be better to be solved for the library itself.

### What did you expect to happen?
The app to run without issues

### What happened instead?
The app crashes

```
dyld[5988]: Symbol not found: _OBJC_CLASS_$_NSError
  Referenced from: <896BA9EF-CAFC-398D-9578-18E4273B0D97> /private/var/containers/Bundle/Application/0EE27EE4-74BB-4956-A320-A148D04A4597/_myApp_
  Expected in:     <26B3444B-5032-35D1-A7D4-BAA506461EC5> /private/var/containers/Bundle/Application/0EE27EE4-74BB-4956-A320-A148D04A4597/_myApp_.app/Frameworks/Ably.framework/Ably
```

### The issue and fix
The issue is here, on the file `ARTRealTimePresence`:
```
- (BOOL)history:(ARTRealtimeHistoryQuery *)query callback:(ARTPaginatedPresenceCallback)callback error:(NSError **)errorPtr {
    query.realtimeChannel = _channel;
    @try {
        return [_channel.restChannel.presence history:query callback:callback error:errorPtr];
    }
    @catch (NSError *error) {
        if (errorPtr) {
            *errorPtr = error;
        }
        return false;
    }
}
```

The 
```
@catch (NSError *error)
```

should not be an NSError but an NSException instead, with a simple change it would work fine again:
```
- (BOOL)history:(ARTRealtimeHistoryQuery *)query callback:(ARTPaginatedPresenceCallback)callback error:(NSError **)errorPtr {
    query.realtimeChannel = _channel;
    @try {
        return [_channel.restChannel.presence history:query callback:callback error:errorPtr];
    }
    @catch (NSException *exception) {
        if (errorPtr) {
            *errorPtr = exception;
        }
        return false;
    }
}
```

**Comment by Roniy08** [comment]
> ### Which version of the Ably SDK are you using?
> 1.2.20
> 
> ### On which platform does the issue happen?
> iOS 17
> 
> ### Are you using Carthage?
> No
> 
> ### Are you using Cocoapods?
> Yes, 1.12.1.
> 
> ### Which version of Xcode are you using?
> (`xcodebuild -version`)
> 
> ### What did you do?
> After installing Xcode 15, just trying to run any app (we have 2 using ably) will make it crash as soon as it starts. I figured, and fixed, the issue, locally, however it would be better to be solved for the library itself.
> 
> ### What did you expect to happen?
> The app to run without issues
> 
> ### What happened instead?
> The app crashes
> 
> ```
> dyld[5988]: Symbol not found: _OBJC_CLASS_$_NSError
>   Referenced from: <896BA9EF-CAFC-398D-9578-18E4273B0D97> /private/var/containers/Bundle/Application/0EE27EE4-74BB-4956-A320-A148D04A4597/_myApp_
>   Expected in:     <26B3444B-5032-35D1-A7D4-BAA506461EC5> /private/var/containers/Bundle/Application/0EE27EE4-74BB-4956-A320-A148D04A4597/_myApp_.app/Frameworks/Ably.framework/Ably
> ```
> 
> ### The issue and fix
> The issue is here, on the file `ARTRealTimePresence`:
> 
> ```
> - (BOOL)history:(ARTRealtimeHistoryQuery *)query callback:(ARTPaginatedPresenceCallback)callback error:(NSError **)errorPtr {
>     query.realtimeChannel = _channel;
>     @try {
>         return [_channel.restChannel.presence history:query callback:callback error:errorPtr];
>     }
>     @catch (NSError *error) {
>         if (errorPtr) {
>             *errorPtr = error;
>         }
>         return false;
>     }
> }
> ```
> 
> The
> 
> ```
> @catch (NSError *error)
> ```
> 
> should not be an NSError but an NSException instead, with a simple change it would work fine again:
> 
> ```
> - (BOOL)history:(ARTRealtimeHistoryQuery *)query callback:(ARTPaginatedPresenceCallback)callback error:(NSError **)errorPtr {
>     query.realtimeChannel = _channel;
>     @try {
>         return [_channel.restChannel.presence history:query callback:callback error:errorPtr];
>     }
>     @catch (NSException *exception) {
>         if (errorPtr) {
>             *errorPtr = exception;
>         }
>         return false;
>     }
> }
> ```

thanks this change of specific code help me resolve error in Xcode 15 beta 2 in iOS 17.0 issue.

**Comment by maratal** [comment]
Fixed in v1.2.21
